python/plugins/processing/algs/gdal/gdal2xyz.py

```diff
@@ -36,6 +37,9 @@
 class gdal2xyz(GdalAlgorithm):
     INPUT = 'INPUT'
     BAND = 'BAND'
+    SRCNODATA = 'SRCNODATA'
+    DSTNODATA = 'DSTNODATA'
+    SKIPNODATA = 'SKIPNODATA'
```

Comment:
```suggestion
    SRCNODATA = 'NODATA_INPUT'
    DSTNODATA = 'NODATA_OUTPUT'
    SKIPNODATA = 'SKIP_NODATA'
```

That would match the naming used elsewhere in the gdal algorithms -- we tend toward uppercase, _ separated strings, where source = INPUT and dest = OUTPUT. See eg the merge algorithm.